File: receptionHubBackend/Controllers/Dtos/RecepcionerDto.cs
using System;
using receptionHubBackend.Models.Enums;

namespace receptionHubBackend.Controllers.Dtos;

public class RecepcionerDto
{
    public int IdRecepcionera { get; set; }
    public string Ime { get; set; } = null!;
    public string Prezime { get; set; } = null!;
    public string KorisnickoIme { get; set; } = null!;
    public string? Email { get; set; }
    public string? BrojTelefona { get; set; }
    public TipPozicije Pozicija { get; set; }
    public bool Aktivan { get; set; }
    public DateTime DatumKreiranjaRacuna { get; set; }
    public DateTime? PosljednjiLogin { get; set; }
    public string? SlikaProfila { get; set; }
    public string? Napomena { get; set; }
}

File: receptionHubFrontend/src/types/dtos/recepcioner.dto.ts
export type TipPozicije = 'Recepcioner' | 'Supervizor' | 'Menadzer' | 'Admin';

export const TipPozicijeValues = {
  Recepcioner: 'Recepcioner' as TipPozicije,
  Supervizor: 'Supervizor' as TipPozicije,
  Menadzer: 'Menadzer' as TipPozicije,
  Admin: 'Admin' as TipPozicije
};

export interface RecepcionerDto {
  idRecepcionera: number;
  ime: string;
  prezime: string;
  korisnickoIme: string;
  email?: string;
  brojTelefona?: string;
  pozicija: TipPozicije;
  aktivan: boolean;
  datumKreiranjaRacuna: string;
  posljednjiLogin?: string;
  slikaProfila?: string;
  napomena?: string;
}

export interface KreirajRecepcioneraDto {
  ime: string;
  prezime: string;
  korisnickoIme: string;
  email?: string;
  brojTelefona?: string;
  pozicija: TipPozicije;
  lozinka: string;
  slikaProfila?: string;
  napomena?: string;
}

export interface UpdateRecepcioneraDto {
  ime?: string;
  prezime?: string;
  email?: string;
  brojTelefona?: string;
  pozicija?: TipPozicije;
  aktivan?: boolean;
  slikaProfila?: string;
  napomena?: string;
}

export interface PromjeniLozinkuDto {
  staraLozinka: string;
  novaLozinka: string;
  potvrdaLozinke: string;
}